writer.go

```diff
@@ -585,15 +607,27 @@ func (w *writer) run() {
 
 	for !done {
 		var mustFlush bool
-
+		if lstMsg != nil {
+			batch = append(batch, lstMsg.msg)
+			resch = append(resch, lstMsg.res)
+			lstMsg = nil
+		}
 		select {
 		case wm, ok := <-w.msgs:
 			if !ok {
 				done, mustFlush = true, true
 			} else {
+				if int(wm.msg.message().size())+currLength > w.maxBytes {
+					// If the size of the current message puts us over the limit,
+					// store the message but don't send it in this batch.
+					mustFlush = true
+					lstMsg = &wm
+					break
```

Comment:
If you change `mustFlush` to represent the number of messages to flush (an `int`), then I think lastMsg and this if-statement go away

Reply:
vowels don't cost anything, I think we'd be ok if we put the `a` back in `lst`